pkg/server/server.go

```diff
@@ -500,7 +509,12 @@ func NewServerWithOpts(opts ...OpenFGAServiceV1Option) (*Server, error) {
 		graph.WithResolveNodeBreadthLimit(s.resolveNodeBreadthLimit),
 	)
 
-	cycleDetectionCheckResolver.SetDelegate(localChecker)
+	trackChecker := graph.NewTrackCheckResolver(
```

Comment:
This should be conditional/optional via config flags. 
This is meant mostly for us to collect metrics as we build the improvements on the caching strategy, so the default OpenFGA binary should not have it enabled unless explicitly configured.

Reply:
This also demands that `ctx` isn't null which requires all the changes instantiating the server.

Reply:
I will add functionality to support it.

Reply:
WithTrackerContext and WithTrackerLogger functions are provided to pass the context.

Reply:
> This also demands that `ctx` isn't null which requires all the changes instantiating the server.
WithTrackerContext and WithTrackerLogger functions are provided to pass the context. Normal startup of the server context already creates and passes context using the aforementioned function.  If your using MustNewServerWithOpts and with environment variable OPENFGA_CHECK_TRACKER_ENABLED = true then yes, you must use function server.WithContext(ctx).